src/identity-providers/add/DetailSettings.tsx

```diff
@@ -141,6 +158,31 @@ export const DetailSettings = () => {
   const isOIDC = providerId.includes("oidc");
   const isSAML = providerId.includes("saml");
 
+  const loader = async () => {
+    const loaderMappers = await adminClient.identityProviders.findMappers({
+      alias,
+    });
+
+    const loaderMapperTypes =
+      await adminClient.identityProviders.findMapperTypes({ alias });
+
+    const components = loaderMappers?.map((loaderMapper) => {
+      const provider = Object.values(loaderMapperTypes).filter(
+        (loaderMapperType) => {
+          return loaderMapper.identityProviderMapper! === loaderMapperType.id!;
+        }
+      ) as unknown as IdPWithMapperAttributes[];
```

Comment:
I believe you could use a `Array#find()` here to get the result you are looking for (as only the first index is ever used).

```suggestion
      const mapperType = loaderMapperTypes.find(
        (loaderMapperType) =>
          loaderMapper.identityProviderMapper! === loaderMapperType.id!
      );
```